gc/gc-base/src/main/java/org/projectnessie/gc/serialization/ImmutableGCParamsSerializer.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.gc.serialization;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.Serializer;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Map;
+import org.projectnessie.gc.base.ImmutableGCParams;
+
+public class ImmutableGCParamsSerializer extends Serializer<ImmutableGCParams> {
+  public ImmutableGCParamsSerializer() {
+    this.setImmutable(true);
+  }
+
+  @Override
+  public void write(Kryo kryo, Output output, ImmutableGCParams immutableGCParams) {
+    kryo.writeClassAndObject(output, immutableGCParams.getNessieClientConfigs());
+    kryo.writeClassAndObject(output, immutableGCParams.getCutOffTimestampPerRef());
```

Comment:
`writeClassAndObject` is used when we need to save a type of the object - it is useful for parameterized types like maps.